tests/test_mc_simulator.py

```diff
@@ -0,0 +1,420 @@
+    ]
+
+    confirm_queue_is_eventually_empty(input_queue)
+    confirm_queue_is_eventually_empty(output_queue)
+    confirm_queue_is_eventually_empty(error_queue)
+    confirm_queue_is_eventually_empty(testing_queue)
+
+
+def test_MantarrayMCSimulator__correctly_stores_time_since_initialized__in_setup_before_loop(
+    mocker,
+    mantarray_mc_simulator_no_beacon,
+):
+    _, _, _, _, simulator = mantarray_mc_simulator_no_beacon
+
+    expected_init_time = 15796649135715
+    expected_poll_time = 15880317595302
+    mocker.patch.object(  # Tanner (2/1/21): mocking perf_counter is generally a bad idea but can't think of any other way to test this
+        mc_simulator,
+        "perf_counter_ns",
+        autospec=True,
+        side_effect=[expected_init_time, expected_poll_time],
+    )
+
+    # before setup
+    assert simulator.get_cms_since_init() == 0
+
+    invoke_process_run_and_check_errors(simulator, perform_setup_before_loop=True)
+    # after setup
+    expected_dur_since_init = (
+        expected_poll_time - expected_init_time
+    ) // NANOSECONDS_PER_CENTIMILLISECOND
+    assert simulator.get_cms_since_init() == expected_dur_since_init
+
+
+def test_MantarrayMCSimulator_read__gets_next_available_bytes(
+    mantarray_mc_simulator_no_beacon,
+):
+    _, _, _, testing_queue, simulator = mantarray_mc_simulator_no_beacon
+
+    expected_bytes = b"expected_item"
+    test_item = {"command": "add_read_bytes", "read_bytes": expected_bytes}
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(
+        test_item, testing_queue
+    )
+    invoke_process_run_and_check_errors(simulator)
+    actual_item = simulator.read(size=len(expected_bytes))
+    assert actual_item == expected_bytes
+
+
+def test_MantarrayMCSimulator_read__returns_empty_bytes_if_no_bytes_to_read(
+    mantarray_mc_simulator,
+):
+    _, _, _, _, simulator = mantarray_mc_simulator
+    actual_item = simulator.read(size=10)
+    expected_item = bytes(0)
+    assert actual_item == expected_item
+
+
+def test_MantarrayMCSimulator_write__puts_object_into_input_queue(
+    mantarray_mc_simulator,
+):
+    input_queue, _, _, _, simulator = mantarray_mc_simulator
+    test_item = b"input_item"
+    simulator.write(test_item)
+    confirm_queue_is_eventually_of_size(input_queue, 1)
+    # Tanner (1/28/21): removing item from queue to avoid BrokenPipeError
+    actual_item = input_queue.get(timeout=QUEUE_CHECK_TIMEOUT_SECONDS)
+    assert actual_item == test_item
+
+
+def test_MantarrayMCSimulator__makes_status_beacon_available_to_read_on_first_iteration__with_random_truncation(
+    mantarray_mc_simulator, mocker
+):
+    spied_randint = mocker.spy(random, "randint")
+
+    _, _, _, _, simulator = mantarray_mc_simulator
+    mocker.patch.object(simulator, "get_cms_since_init", autospec=True, return_value=0)
+
+    invoke_process_run_and_check_errors(simulator)
+    spied_randint.assert_called_once_with(0, 10)
+
+    timestamp_bytes = bytes(8)
```

Comment:
to make this a bit more clear (for example if we're looking back months from now during a refactor), I think this relates to mocking in line 180 right?  so there could be a variable `cms_since_init` and then you can set the `return_value` to that and then also say `timestamp_bytes = cms_since_init.to_bytes(8,byte_order='little')`